apps/web-app/pages/redirect.vue

```diff
@@ -1,12 +1,20 @@
 <script setup lang="ts">
-  import { navigateTo, useNuxtApp, onMounted } from '#imports';
+  import { navigateTo, useNuxtApp, onMounted, useCookie } from '#imports';
 
   const { $trpc } = useNuxtApp();
 
   onMounted(async () => {
-    const { defaultOrgShortcode, twoFactorEnabledCorrectly } =
+    const { defaultOrgShortcode, twoFactorEnabledCorrectly, username } =
       await $trpc.account.defaults.redirectionData.query({});
 
+    const usernameCookie = useCookie('un-join-username', {
+      maxAge: 3600
+    });
+
+    if (process.client && !usernameCookie.value && username) {
+      usernameCookie.value = username;
+    }
+
     if (!twoFactorEnabledCorrectly) {
       return navigateTo(`/join/2fa`);
```

Comment:
this may be a problem if they already have an org set up
we should set a cookie or something so the join/2fa page will redirect back to org when done

Reply:
check if the user have org already, if they have a org skip it.

Reply:
maybe, but then they wont set up 2fa again

Reply:
ah, do you have a better plan?